optional/something.py

```diff
@@ -4,6 +4,7 @@
 
 class Something(AbstractOptional):
     def __init__(self, value, optional):
+        assert value is not None
```

Comment:
The problem with raw asserts is that the message is not as explicit as it could be.  A custom exception could be more specific.  If you'd rather not though, I will let this through.

Reply:
No, I was being lazy. I'll change this before we merge.

Reply:
I opted for a ValueError. If you strongly desire a custom exception, I'll have it derive from ValueError.